meshtastic/interfaces/ble/connection.py

```
@@ -0,0 +1,472 @@
+                try:
+                    direct_timeout = min(
+                        DIRECT_CONNECT_TIMEOUT_SECONDS, BLEConfig.CONNECTION_TIMEOUT
+                    )
+                    self.client_manager.connect_client(client, timeout=direct_timeout)
+                    self._finalize_connection(
+                        client,
+                        target_address,
+                        register_notifications_func,
+                        on_connected_func,
+                    )
+                    return client
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)